powerstore/provider.go

```diff
@@ -2,68 +2,236 @@ package powerstore
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"strconv"
+	client "terraform-provider-powerstore/client"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
+// Ensure provider defined types fully satisfy framework interfaces
+var _ tfsdk.Provider = &Pstoreprovider{}
+
 // Provider -
-func Provider() *schema.Provider {
-	return &schema.Provider{
-		Schema: map[string]*schema.Schema{
-			"host": &schema.Schema{
-				Type:        schema.TypeString,
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("POWERSTORE_HOST", nil),
+type Pstoreprovider struct {
+	// client can contain the upstream provider SDK or HTTP client used to
+	// communicate with the upstream service. Resource and DataSource
+	// implementations can then make calls using this client.
+	//
+	client client.Client
+
+	// configured is set to true at the end of the Configure method.
+	// This can be used in Resource and DataSource implementations to verify
+	// that the provider was previously configured.
+	configured bool
+
+	// version is set to the provider version on release, "dev" when the
+	// provider is built and ran locally, and "test" when running acceptance
+	// testing.
+	version string
+}
+
+// ProviderData can be used to store data from the Terraform configuration.
+type ProviderData struct {
+	Endpoint types.String `tfsdk:"endpoint"`
+	Insecure types.Bool   `tfsdk:"insecure"`
+	Password types.String `tfsdk:"password"`
+	Username types.String `tfsdk:"username"`
+}
+
+func (p *Pstoreprovider) Configure(ctx context.Context, req tfsdk.ConfigureProviderRequest, resp *tfsdk.ConfigureProviderResponse) {
+	// If the upstream provider SDK or HTTP client requires configuration, such
+	// as authentication or logging, this is a great opportunity to do so.
+	// TODO: Implement client using schema
+	config := ProviderData{}
+
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+
+	// User must provide a user to the provider
+	if config.Username.Unknown {
+		// Cannot connect to client with an unknown value
+		resp.Diagnostics.AddWarning(
+			"Unable to create client",
+			"Cannot use unknown value as username",
+		)
+		return
+	}
+
+	if config.Username.Value == "" {
```

Comment:
Please check if we need env variable other than acceptance testing

Reply:
Removed env variables